lib/internal/modules/cjs/loader.js

```diff
@@ -121,15 +121,39 @@ Module._extensions = Object.create(null);
 var modulePaths = [];
 Module.globalPaths = [];
 
-Module.wrap = function(script) {
+Module.patched = false;
+
+Module._wrap = function(script) {
   return Module.wrapper[0] + script + Module.wrapper[1];
 };
 
-Module.wrapper = [
+Module._wrapper = [
   '(function (exports, require, module, __filename, __dirname) { ',
   '\n});'
 ];
 
+Object.defineProperty(Module, 'wrap', {
+  get: function() {
+    return Module._wrap;
+  },
+
+  set: function(value) {
+    Module.patched = true;
+    Module._wrap = value;
+  }
+});
+
+Object.defineProperty(Module, 'wrapper', {
+  get: function() {
+    return Module._wrapper;
+  },
+
+  set: function(value) {
+    Module.patched = true;
+    Module._wrapper = value;
+  }
+});
+
 const debug = util.debuglog('module');
```

Comment:
Could you go a step further and add the `set` and `patched` trap on the index `0` and `1` of the `wrapper` array. This may be a good time to DRY out the patched `set`/`get` pattern too.

Reply:
might be easier / more robust to just wrap things in a Proxy to see any mutation including Object.defineOwnProperty

Reply:
@bmeck Yep, that would be great too!